### PR TITLE
Use home project as namespace for branched package

### PR DESCRIPTION
**Pull request by vpereira**
As requested, namespace `target_project` under `home:$USER` 

At least for testing, to create the home project was necessary.

To test, I'm setting the `User.session!` to `user`, however this information should come from `token.user`

 
TODO:
- [x] check dependency on home project
- [ ] pass `token.user` to the `BranchPackageStep.call` method
https://trello.com/c/F014licj/1688-place-the-branched-package-under-home

### CODE REVIEW

**Comment by vpereira** [comment]
[Place the branched package under home:](https://trello.com/c/F014licj/1688-place-the-branched-package-under-home)

**Comment by saraycp** [comment]
> To test, I'm setting the `User.session!` to `user`, however this information should come from `token.user`

After #11221  gets merged, the `token` will be passed to the `BranchPackageStep` model, so you can use `@token.user` instead of the session.

File: src/api/app/models/workflow/step/branch_package_step.rb [hunk 1]
```diff
@@ -42,7 +42,9 @@ def target_package
       end
 
       def target_project
-        source_project + ":PR-#{@scm_extractor_payload[:pr_number]}"
+        # TODO
+        # it must be token.user.login
```
Comment (hennevogel) [inline src/api/app/models/workflow/step/branch_package_step.rb:46]: *nitpick* 
```suggestion
        # TODO: should be token.user.login
```

**Review by hennevogel — APPROVED** [review]

**Review by saraycp — APPROVED** [review]
LGTM. Let's try to merge and rebase PR #11221 first.

**Pull request merged**